### PR TITLE
Add HVQFN-40

### PR DESCRIPTION
**Pull request by jneiva08**
Created footprint for HVQFN 40

Datasheet: https://www.nxp.com/docs/en/package-information/SOT618-1.pdf
and https://www.nxp.com/docs/en/nxp/data-sheets/PN532_C1.pdf

### CODE REVIEW

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/hvqfn.yaml [hunk 1]
```diff
@@ -107,3 +107,59 @@ HVQFN-24-1EP_4x4mm_P0.5mm_EP2.6x2.6mm:
...
+  overall_height:
+    minimum: 0.8
+    nominal: 0.85
+    maximum: 1
+  lead_width:
+    minimum: 0.18
+    maximum: 0.3
+  lead_len:
+    minimum: 0.3
+    maximum: 0.5
```
Comment (cpresser) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/hvqfn.yaml:135]: nominal: 0.4

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/hvqfn.yaml [hunk 2]
```diff
@@ -107,3 +107,59 @@ HVQFN-24-1EP_4x4mm_P0.5mm_EP2.6x2.6mm:
...
+    minimum: 5.9
+    nominal: 6.0
+    maximum: 6.1
+  overall_height:
+    minimum: 0.8
+    nominal: 0.85
+    maximum: 1
+  lead_width:
+    minimum: 0.18
+    maximum: 0.3
```
Comment (cpresser) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/hvqfn.yaml:132]: nominal: 0.21

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/hvqfn.yaml [hunk 3]
```diff
@@ -107,3 +107,59 @@ HVQFN-24-1EP_4x4mm_P0.5mm_EP2.6x2.6mm:
   #pad_length_addition: 0.5
   #suffix: '_Pad{pad_x:.2f}x{pad_y:.2f}mm_HandSolder'
   #include_suffix_in_3dpath: 'False'
+
+HVQFN-40-1EP_6x6mm_P0.5mm_EP4.25x4.25mm:
```
Comment (cpresser) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/hvqfn.yaml:111]: The nominal EP_size is 4.1 which is also in the output of the generator.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/hvqfn.yaml [hunk 4]
```diff
@@ -107,3 +107,59 @@ HVQFN-24-1EP_4x4mm_P0.5mm_EP2.6x2.6mm:
...
+  EP_size_x:
+    minimum: 3.95
+    maximum: 4.25
+  EP_size_y:
+    minimum: 3.95
+    maximum: 4.25
+  # EP_paste_coverage: 0.65
+  EP_num_paste_pads: [3, 3]
+  #EP_size_limit_x: 2.7  #for relatively large EP pads (increase clearance)
+  #EP_size_limit_y: 2.7  #for relatively large EP pads (increase clearance)
```
Comment (cpresser) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/hvqfn.yaml:146]: The recommended footprint has way less paste than the generated pads:
image
image

TBH, I am not sure what to do here. @poeschlr can you help?

**Review by cpresser — CHANGES_REQUESTED** [review]

**Comment by jneiva08** [comment]
@cpresser have made the modifications you requested, except the last one
I don't know too. let me know what should change.

**Comment by cpresser** [comment]
I played around a little with the parameters. To get the paste-pads exactly the size of the datasheet one has to enter `19.2742%` (calculated by 9*0.6^2/4.1^2).
Getting the paste pads onto the correct position only works with the thermal_vias variant when `grid: [0.9, 0.9]` is used.

**But that is all useless.** This is supposed to be an IPC compliant footprint. Not a vendor specific one. It took me a while to come to that obvious conclusion.
Thus, you can just have the default values for pad geometry.

I just have one more wish, please remove the commented lines. They always look like they are there for a reason.  But they are not which is bamboozling. Its just copy-paste leftovers. While you are at it, also remove them for the other two parts in that file.

**Comment by myfreescalewebpage** [comment]
@jneiva08 ping, can you follow the query of @cpresser (removing comment lines in the yaml file)

**Comment by jneiva08** [comment]
@myfreescalewebpage @cpresser 
When you mean remove comment lines you mean:
`#manufacturer
#part_number
#ipc_density
#EP_paste_coverage
#EP_size_limit_x
#EP_size_limit_y
#min_annular_ring
#EP_num_paste_pads
#grid
#bottom_pad_size
#paste_avoid_via
#chamfer_edge_pins
#pin_count_grid
#pad_length_addition
#suffix
#include_suffix_in_3dpath`

**Comment by myfreescalewebpage** [comment]
@jneiva08 yes

**Comment by jneiva08** [comment]
@myfreescalewebpage @cpresser, have made the changes have requested, hope didn't break the file. Need to make more changes?

**Comment by cpresser** [comment]
Thanks, that looks great.
The merge-commit was not needed, but its not a problem either. I am going to squash this branch before merging it to master anyway.

Can you please update the footprint PR (https://github.com/KiCad/kicad-footprints/pull/2303) with the latest generator output (just the HVQFN-40) file. Then I will merge that as well.

**Review by cpresser — APPROVED** [review]

**Comment by jneiva08** [comment]
Thanks for your time, I will soon update in kicad-footprints

**Pull request merged**